tests/api/test_virtual.py

```diff
@@ -77,11 +78,11 @@ def example_product(name):
     pixelquality = dict(name='pixelquality', dtype='int16', nodata=0, units='1',
                         flags_definition=flags)
 
-    result = DatasetType(example_metadata_type(),
-                         dict(name=name, description="", metadata_type='eo', metadata={}))
+    result = Product(example_metadata_type(),
+                     dict(name=name, description="", metadata_type='eo', metadata={}))
     result.grid_spec = GridSpec(crs=CRS('EPSG:3577'),
-                                tile_size=(100000., 100000.),
-                                resolution=(-25, 25))
+                                tile_shape=(100000, 100000),
+                                resolution=25)
```

Comment:
new gridspec takes pixels not meters:

```suggestion
                                tile_shape=(4000, 4000),
                                resolution=25)
```